Makefile

```diff
@@ -1,72 +1,101 @@
+default: test
+
+# All builds
+
+build: build-rust build-kotlin
+
+build-rust:
+	cargo build --all
+
+build-kotlin:
+	./gradlew build -x test
+
+build-swift:
+	bin/run-ios-build.sh
+
 build-apk:
 	./gradlew glean-core:build
 	./gradlew glean-sample-app:build
-.PHONY: build-apk
 
-install:
-	$(ANDROID_HOME)/platform-tools/adb install -r ./samples/android/app/build/outputs/apk/debug/glean-sample-app-debug.apk
-.PHONY: install
+.PHONY: build build-rust build-kotlin build-swift build-apk
 
-emulator:
-	$(ANDROID_HOME)/emulator/emulator -avd Nexus_5X_API_P -netdelay none -netspeed full
-.PHONY: install
+# All tests
+
+test: test-rust test-kotlin
+
+test-rust:
+	cargo test --all
 
-clippy: fmt
+test-rust-with-logs:
+	RUST_LOG=glean_core=debug cargo test --all -- --nocapture --test-threads=1
+
+test-kotlin:
+	./gradlew test
+
+test-swift:
+	bin/run-ios-tests.sh
+
+.PHONY: test test-rust test-rust-with-logs test-kotlin test-swift
+
+# Linting
+
+lint: clippy ktlint
+
+clippy:
 	cargo clippy --all --all-targets --all-features -- -D warnings
-.PHONY: lint
 
 ktlint:
 	./gradlew ktlint detekt
-.PHONY: lint
-
-lint: clippy ktlint
-.PHONY: lint
 
 swiftlint:
 	swiftlint --strict
-	swiftformat glean-core/ios samples/ios --swiftversion 5 --verbose --lint
-.PHONY: swiftlint
 
-fmt:
+.PHONY: lint clippy ktlint swiftlint
+
+# Formatting
+
+fmt: rustfmt
+
+rustfmt:
 	cargo fmt --all
-.PHONY: fmt
 
 swiftfmt:
 	swiftformat glean-core/ios samples/ios --swiftversion 5 --verbose
-.PHONY: swiftfmt
-
-test:
-	cargo test --all
-	./gradlew test
-.PHONY: test
 
-test-rust-with-logs:
-	RUST_LOG=glean_core=debug cargo test --all -- --nocapture --test-threads=1
-.PHONY: test-rust-with-logs
+.PHONY: fmt rustfmt swiftfmt
 
-test-ios:
-	./bin/run-ios-tests.sh
+# Docs
 
-cbindgen:
-	RUSTUP_TOOLCHAIN=nightly \
-	cbindgen glean-core/ffi --lockfile Cargo.lock -o glean-core/ffi/glean.h
-	cp glean-core/ffi/glean.h glean-core/ios/Glean/GleanFfi.h
-.PHONY: cbindgen
+docs: rust-docs kotlin-docs
 
-docs:
+rust-docs:
 	bin/build-rust-docs.sh
+
+kotlin-docs:
 	./gradlew docs
-.PHONY: docs
 
-docs-swift:
+swift-docs:
 	bin/build-swift-docs.sh
-.PHONY: docs
+
+.PHONY: docs rust-docs kotlin-docs swift-docs
 
 linkcheck: docs
```

Comment:
Strictly speaking, I guess the only way to make this work is to also build `swift-docs`, but I think that's only iOS.  (That's kind of a pain -- we might have to modify the linkchecker config so it thinks that any links into the iOS docs are always valid...)

Reply:
*secretly adds a header saying you have to use macos to do glean*

Reply:
I can't run the linkchecker locally anyway (because ... reasons), so ... I trust CI to have it all.

Reply:
Yeah -- at the end of the day, as long as it works on CI, that's all that really matters.  This isn't a complex thing like a compiler where it's critical to be able to test things locally.